configuration/developing-community-health-applications.md

```diff
@@ -366,6 +366,18 @@ function isFormFromArraySubmittedInWindow(reports, formsArray, startTime, endTim
 }
 ```
 
+#### What to put in appliesIf vs resolvedIf
+Both functions `appliesIf` and `resolvedIf` capture logic for when your task should appear to the user. So why are there two functions which seemingly result in the same behaviour? The answer is that once a task appears (`appliesIf: true`, `resolvedIf: false`), that task will not disappear until it is explicitly told to resolve (`appliesIf: true`, `resolvedIf: true`). If you're seeing your task linger when you expect it to disappear, but it does disappear when you reload the page (refresh the browser) - you probably need to move logic from `appliesIf` into `resolvedIf`.
+
+Generally:
+
+* `appliesIf` should contain your immutable expectations for the task - conditions that are needed for the task to appear, but that you don't expect to change once they are met. If this function returns `true` today and then returns `false` tomorrow for the same input, it will block you from correctly resolving the emitted tasks.
+* `resolvedIf` should capture the change in state which (in addition to the logic in appliesIf being true) causes the task to resolve.
+
+So why not put everything in `resolvedIf` and this will always work without thinking? You can. But there are negative performance implications of doing this, and you should endeavour to capture all that you can in `appliesIf` without breaking things. This performance concern is particularly important for tasks with `appliesTo: 'reports'`.
+
+Checking if a patient is younger than 5 years of age, or logic to test if a contact has form x existing before any form y - these are examples of bad things to put in `appliesIf` functions.
```

Comment:
I would have thought that screening for patient age would have been a good thing to have in the `appliesIf` function.

Reply:
If you're checking somebody is younger than x, if they get too old while the task is present, it won't be able to be cleared. I guess it is reasonable to have it in appliesIf. Not a great example.